router/core/operation_processor.go

```diff
@@ -363,30 +448,182 @@ func (o *OperationKit) Normalize() error {
+	o.parsedOperation.Type = entry.operationType
+	if o.parsedOperation.Request.Variables == nil || bytes.Equal(o.parsedOperation.Request.Variables, literalNull) {
+		o.parsedOperation.Request.Variables = []byte("{}")
+	}
+
+	err = jsonparser.ObjectEach(entry.exportedVariables, func(key []byte, value []byte, dataType jsonparser.ValueType, offset int) (err error) {
+		_, exists := o.parsedOperation.VariablesMap[string(key)]
+		if exists {
+			return nil
+		}
+		if dataType == jsonparser.String {
+			stringValue := make([]byte, len(value)+2)
+			stringValue[0] = '"'
+			copy(stringValue[1:], value)
+			stringValue[len(stringValue)-1] = '"'
+			o.parsedOperation.Request.Variables, err = jsonparser.Set(o.parsedOperation.Request.Variables, stringValue, unsafebytes.BytesToString(key))
+		} else {
+			o.parsedOperation.Request.Variables, err = jsonparser.Set(o.parsedOperation.Request.Variables, value, unsafebytes.BytesToString(key))
+		}
+		return
+	})
+	if err != nil {
+		return false, errors.WithStack(fmt.Errorf("failed to set default values for variables: %w", err))
+	}
+
+	o.kit.doc.Input.ResetInputString(entry.normalizedRepresentation)
+	report := &operationreport.Report{}
+	o.kit.parser.Parse(o.kit.doc, report)
+	if report.HasErrors() {
+		return false, &reportError{
+			report: report,
+		}
+	}
+
+	return true, nil
+}
+
+func (o *OperationKit) savePersistedOperationToCache(skipIncludeVariableNames []string, exportedVariables []byte) {
+	if o.cache == nil {
+		return
+	}
+	cacheKey := o.generatePersistedOperationCacheKey(skipIncludeVariableNames)
+
+	entry := normalizedOperationCacheEntry{
+		operationID:              o.parsedOperation.ID,
+		normalizedRepresentation: o.parsedOperation.NormalizedRepresentation,
+		operationType:            o.parsedOperation.Type,
+		exportedVariables:        make([]byte, len(exportedVariables)),
+	}
+	copy(entry.exportedVariables, exportedVariables)
+
+	o.cache.persistedOperationCacheLock.Lock()
+	o.cache.persistedOperationCache[cacheKey] = entry
+	o.cache.persistedOperationCacheLock.Unlock()
+
+	o.cache.persistetOperationVariableNamesLock.Lock()
+	o.cache.persistetOperationVariableNames[o.parsedOperation.GraphQLRequestExtensions.PersistedQuery.Sha256Hash] = skipIncludeVariableNames
+	o.cache.persistetOperationVariableNamesLock.Unlock()
+}
+
+var (
+	literalF    = []byte("f")
+	literalT    = []byte("t")
+	literalNull = []byte("null")
+)
+
+func (o *OperationKit) loadPersistedOperationCacheKey(persistedQuerySha256Hash string) (key uint64, ok bool) {
+	o.cache.persistetOperationVariableNamesLock.RLock()
+	variableNames, ok := o.cache.persistetOperationVariableNames[persistedQuerySha256Hash]
+	o.cache.persistetOperationVariableNamesLock.RUnlock()
+	if !ok {
+		return 0, false
+	}
+	key = o.generatePersistedOperationCacheKey(variableNames)
+	return key, true
+}
+
+func (o *OperationKit) generatePersistedOperationCacheKey(skipIncludeVariableNames []string) uint64 {
+	o.kit.keyGen.Reset()
```

Comment:
A minor thing but IMO we should always clear memory after a task otherwise the memory of the last task is preserved for no reason.